src/nvim/api/win_config.c

```diff
@@ -603,6 +697,25 @@ static bool parse_float_config(Dict(float_config) *config, FloatConfig *fconfig,
     return false;
   }
 
+  if (HAS_KEY(config->title_pos)) {
+    if (!HAS_KEY(config->title)) {
+      api_set_error(err, kErrorTypeException, "title_pos must set with title");
+      return false;
+    }
+  }
+
+  if (HAS_KEY(config->title)) {
+    // title only work with border
+    if (!HAS_KEY(config->border)) {
+      api_set_error(err, kErrorTypeException, "title must set with border");
```

Comment:
`test/functional/ui/float_spec.lua` should have a `it('validates')` test. If it doesn't, we need one. At least to cover a few of these, to exercise the code.

Reply:
okay will add this testcase.